react/src/lib/components/SubsurfaceViewer/layers/labeled_points/labeledPointsLayer.stories.tsx

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import { ComponentStory, ComponentMeta } from "@storybook/react";
+
+import SubsurfaceViewer from "../../SubsurfaceViewer";
+import { default as LabeledPointsLayer } from "./labeledPointsLayer";
+import { default as AxesLayer } from "../axes/axesLayer";
+
+export default {
+    component: SubsurfaceViewer,
+    title: "SubsurfaceViewer / LabeledPoints Layer",
+} as ComponentMeta<typeof SubsurfaceViewer>;
+
+const defaultParameters = {
+    docs: {
+        inlineStories: false,
+        iframeHeight: 500,
+    },
+};
+
+// Small example using labeledPointsLayer.
+const labeledPointsLayer = new LabeledPointsLayer({
+    "@@type": "LabeledPointsLayer",
+    id: "labeledPoints-layer",
+
+    /*eslint-disable */
+    pointsData:   [  0,  0,  5,  // Vertex 1, x, y, z
+                    10,  0,  5,  // Vertex 2, x, y, z
+                    10, 10,  5,  // ...
+                     0, 10,  0,
+                     5, -5, 10,
+                    11, -4,  6,
+                    11,  0,  7,
+                    17,  0,  8
+                    ],    
+    /*eslint-enable */
+    color: [255, 100, 100],
+    pointRadius: 10,
+    radiusUnits: "pixels",
+    ZIncreasingDownwards: true,
+});
+
+const smallAxesLayer = new AxesLayer({
+    "@@type": "AxesLayer",
+    id: "axes_small",
+    bounds: [-10, -10, 0, 20, 10, 10],
+});
+
+export const SmallLabeledPointsLayer: ComponentStory<
+    typeof SubsurfaceViewer
+> = (args) => {
+    return <SubsurfaceViewer {...args} />;
+};
+
+SmallLabeledPointsLayer.args = {
+    id: "map",
+    layers: [smallAxesLayer, labeledPointsLayer],
+    bounds: [-10, -10, 17, 10],
+    views: {
+        layout: [1, 1],
+        viewports: [
+            {
+                id: "view_1",
+                show3D: true,
+            },
+        ],
+    },
+};
+
+SmallLabeledPointsLayer.parameters = {
+    docs: {
+        ...defaultParameters.docs,
+        description: {
+            story: "Point coordinates are given as native javascript array.",
+        },
+    },
+};
+
+// Huge example using labeledPointsLayer.
+const sideSize = 10000;
+const pointsCount = 100000;
+
+const hugePointsLayer = new LabeledPointsLayer({
+    "@@type": "LabeledPointsLayer",
+    id: "labeledPoints-layer",
+    pointsData: Array(pointsCount * 3)
+        .fill(0)
+        .map(() => Math.random() * sideSize),
```

Comment:
Using random coordinates would prevent the ability for snapshot based testing. Suggest using a seeded PRNG or another deterministic algorithm to generate the coordinates.

Reply:
There are two stories for every layer: the first one has some hardcoded data so suitable for snapshot testing, another uses huge randomly generated data for stress testing and revealing performance issues. It isn't intended for snapshot testing.

Reply:
I think it could be relevant also for performance testing to have deterministic rendering, as one way to determine if a canvas based scene has completed rendering, in order to measure time, is to retry-wait using snapshot comparison.

Also, this might prevent us from using some of the automatic testing tools from storybook, which I think would be worth checking out in the future.

Moreover, it should be straightforward to replace Math.random() with a seedable PRNG.